packages/ra-ui-materialui/src/button/SaveButton.tsx

```diff
@@ -64,28 +70,45 @@ export const SaveButton = <RecordType extends RaRecord = any>(
         ...rest
     } = props;
     const translate = useTranslate();
-    const form = useForm();
+    const notify = useNotify();
+    const form = useFormContext();
     const saveContext = useSaveContext();
     const hasSideEffects = !!mutationOptions || !!transform;
     const type = !submitOnEnter || hasSideEffects ? 'button' : 'submit';
 
-    const handleClick: MouseEventHandler<HTMLButtonElement> = event => {
-        if (onClick) {
-            onClick(event);
-        }
-        if (event.defaultPrevented) {
-            return;
-        }
-
-        if (hasSideEffects) {
-            event.preventDefault();
-            const values = form.getState().values;
-            saveContext?.save(values, {
-                ...mutationOptions,
-                transform,
-            });
-        }
-    };
+    const handleClick: MouseEventHandler<HTMLButtonElement> = useCallback(
+        async event => {
+            if (onClick) {
+                onClick(event);
+            }
+            if (event.defaultPrevented) {
+                return;
+            }
+            if (hasSideEffects) {
+                event.preventDefault();
+                const isFormValid = await form.trigger();
```

Comment:
I think this logic should be in the form rather than in the SaveButton. Otherwise, will it be triggered when users submit the form using the Enter key? Or when using another Save button?

Reply:
Nope, it's also in the Form but has to be there too for SaveButton that are not submit